mzLib/Transcriptomics/Classes/Nucleotide.cs

```diff
@@ -0,0 +1,232 @@
+        /// </summary>
+        internal static readonly Dictionary<string, Nucleotide> AllKnownResidues;
+
+        internal static readonly Nucleotide[] ResiduesByLetter;
+
+        /// <summary>
+        /// Constructs all known nucleic acids
+        /// </summary>
+        static Nucleotide()
+        {
+         
+            AllKnownResidues = new Dictionary<string, Nucleotide>(66);
+            ResiduesByLetter = new Nucleotide['z' + 1]; //Make it big enough for all the Upper and Lower characters
+
+            // actual base chemical formula after bonding with the sugar
+            // the sugar and phosphate has a chemical formula of C5H8O6P1
+            AdenineBase = AddResidue("Adenine", 'A', "Ade", "C5H4N5");
+            CytosineBase = AddResidue("Cytosine", 'C', "Cyt", "C4H4N3O1");
+            GuanineBase = AddResidue("Guanine", 'G', "Gua", "C5H4N5O1");
+            UracilBase = AddResidue("Uracil", 'U', "Ura", "C4H3N2O2");
+
+            // DNA bases which have the same mass as the ones above
+            // however, naming to deoxy- to distinguish DNA nucleotide mass calculation from RNA
+            DeoxyAdenineBase = AddResidue("DeoxyAdenine", 'B', "dAde", "C5H4N5");
+            DeoxyCytosineBase = AddResidue("DeoxyCytosine", 'D', "dCyt", "C4H4N3O1");
+            DeoxyGuanineBase = AddResidue("DeoxyGuanine", 'H', "dGua", "C5H4N5O1");
+            DeoxyThymineBase = AddResidue("DeoxyThymine", 'V', "dThy", "C5H5N2O2");
+        }
+
+        /// <summary>
+        /// Adds residue to static internal dictionary of all known residues
+        /// </summary>
+        /// <param name="residue"></param>
+        private static void AddResidueToDictionary(Nucleotide residue)
+        {
+            AllKnownResidues.Add(residue.Letter.ToString(CultureInfo.InvariantCulture), residue);
+            AllKnownResidues.Add(residue.Name, residue);
+            AllKnownResidues.Add(residue.Symbol, residue);
+            ResiduesByLetter[residue.Letter] = residue;
+            ResiduesByLetter[Char.ToLower(residue.Letter)] = residue;
+        }
+
+        #endregion
+
+        #region Properties
+
+        private ChemicalFormula _sugarAndPhosphate;
+
+        public string Name { get; private set; }
+        public double MonoisotopicMass { get; private set; }
+        public ChemicalFormula ThisChemicalFormula { get; private set; }
+        public ChemicalFormula BaseChemicalFormula { get; private set; }
+        public char Letter { get; private set; }
+        public string Symbol { get; private set; }
+
+        #endregion
+
+        #region Constructors
+
+        internal Nucleotide(string name, char oneLetterAbbreviation, string threeLetterAbbreviation, string baseChemicalFormula)
+            : this(name, oneLetterAbbreviation, threeLetterAbbreviation, ChemicalFormula.ParseFormula(baseChemicalFormula))
+        {
+        }
+
+        public Nucleotide(string name, char oneLetterAbbreviation, string threeLetterAbbreviation,
```

Comment:
Can you add comments making it clear that this constructor can throw an Exception if an invalid sequence is passed in?